### PR TITLE
opentsdb 2.4.1

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by bayandin** [comment]
Do we want to try to build version 2.4.1? It has been released after we deprecated the formula: https://github.com/OpenTSDB/opentsdb/releases/tag/v2.4.1

**Comment by chenrui333** [comment]
yeah, we can give that a shot.

**Comment by SMillerDev** [comment]
configure needs to be generated with these archives

**Comment by cho-m** [comment]
This won't build with JDK11 or later. Can try adding to `versioned_dependencies_conflicts_allowlist.json` and build with JDK8.

**Comment by cho-m** [comment]
Need to figure out Monterey and Linux test failure (may be hbase LZO native library).

May be able to remove from audit list if we only use `openjdk@8` for build and allow `openjdk@11` at runtime, but will need to make sure runtime is compatible.

File: Formula/opentsdb.rb [hunk 1]
```diff
@@ -16,30 +16,36 @@ class Opentsdb < Formula
...
-           "--disable-silent-rules",
-           "--prefix=#{prefix}",
-           "--mandir=#{man}",
-           "--sysconfdir=#{etc}",
-           "--localstatedir=#{var}/opentsdb"
-    system "make"
-    bin.mkpath
-    (pkgshare/"static/gwt/opentsdb/images/ie6").mkpath
-    system "make", "install"
+    with_env(JAVA_HOME: Language::Java.java_home("1.8")) do
```
Comment (cho-m) [inline Formula/opentsdb.rb:29]: Need to check if `JAVA_HOME` is actually detected in configure scripts. Mainly for Linux which may end up arbitrarily picking OpenJDK version. macOS will use `/usr/bin/java` which is `JAVA_HOME`-aware.

Comment (danielnachun) [reply Formula/opentsdb.rb:29]: I don't have a solution for this yet, but I've seen an identical failure for `drake`, `hexgui`, and `voldemort` on Linux as well, all of which also use `openjdk@8` as well.  For those 3 formulae, the test works locally but fails in CI, so there must be some issue with how we are setting `JAVA_HOME` for some `openjdk@8` bottles.  

I downloaded the Linux bottle and I can see that the value we set `JAVA_HOME` to is `@@HOMEBREW_JAVA@@`.  That is a placeholder that will be replaced when the bottle is poured, and I'm not entirely sure how it is determined.  But if it is not set to the right value then that would explain why the test fails.

Comment (cho-m) [reply Formula/opentsdb.rb:29]: One common issue for Linux is that some formulae have scripts that run `java` from `PATH`, which only works with linked `openjdk` formula. The usual fix I use is the write env script with `PATH=$JAVA_HOME/bin:$PATH`.

There is another issue on Linux test where hbase commands are failing with `ERROR: KeeperErrorCode = ConnectionLoss for /hbase/master`. I haven't had chance to figure out why.

**Comment by cho-m** [comment]
Noting that the current blocker is figuring out why `hbase` LZO is broken on Monterey. I added a test in #104233 that hits this issue, which is reason why `opentsdb` is failing on Monterey.

Alternatively, could drop LZO and use uncompressed or GZ instead.

**Comment by SMillerDev** [comment]
> Alternatively, could drop LZO and use uncompressed or GZ instead.

Let's do that since nobody had a better idea

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**